src/System.Security.Cryptography.Encryption/src/System/Security/Cryptography/CryptoStream.cs

```diff
@@ -0,0 +1,456 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Diagnostics.Contracts;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Security.Cryptography
+{
+    [System.Runtime.InteropServices.ComVisible(true)]
+    public class CryptoStream : Stream, IDisposable
+    {
+        // Member veriables
+        private readonly Stream _stream;
+        private readonly ICryptoTransform _Transform;
+        private byte[] _InputBuffer;  // read from _stream before _Transform
+        private int _InputBufferIndex = 0;
```

Comment:
Nit: we don't need to initialize these to their default values (or if we want to for some reason, we should be consistent)